src/crimson/mgr/client.cc

```diff
@@ -61,10 +61,9 @@ seastar::future<> Client::ms_dispatch(crimson::net::Connection* conn,
 seastar::future<> Client::ms_handle_reset(crimson::net::ConnectionRef c)
 {
   if (conn == c) {
-    return reconnect();
-  } else {
-    return seastar::now();
+    reconnect();
```

Comment:
thanks for the detailed analysis. it explain the root cause well!

but i don't quite like the way how the issue is fixed. because, without the protection of the `pending_dispatch` , the dispatcher could be a dangling pointer when the messenger is trying to access it.

instead, can we 

1. rename `report_timer` to `tick_timer`
1. reset `conn` when the connection is reset.
1. check `conn` for null and reconnect if necessary in `tick_timer` 's callback. report to mgr if `conn` is not null in that callback.

? what do you think?

Reply:
I've modified the code as you suggested:-) Please take a look:-)